backend/api/Repositories/MayanDocumentRepository.cs

```diff
@@ -50,6 +50,87 @@ public class MayanDocumentRepository : IDocumentRepository
+
+            _logger.LogDebug("Retrieving document types...");
+            using HttpClient client = _httpClientFactory.CreateClient("Pims.Api.Logging");
+            client.DefaultRequestHeaders.Accept.Clear();
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Token", authenticationToken);
+            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue(MediaTypeNames.Application.Json));
+
+            Dictionary<string, string> queryParams = new();
+
+            if (!string.IsNullOrEmpty(ordering))
+            {
+                queryParams["ordering"] = ordering;
+            }
+
+            if (page.HasValue)
+            {
+                queryParams["page"] = page.ToString();
+            }
+
+            if (pageSize.HasValue)
+            {
+                queryParams["page_size"] = pageSize.ToString();
+            }
+
+            try
+            {
+                string endpointString = $"{this._config.BaseUri}/document_types/";
+                Uri endpoint = new(QueryHelpers.AddQueryString(endpointString, queryParams));
+                HttpResponseMessage response = await client.GetAsync(endpoint).ConfigureAwait(true);
+                string payload = await response.Content.ReadAsStringAsync().ConfigureAwait(true);
+                this._logger.LogTrace("Response: {response}", response);
+                switch (response.StatusCode)
+                {
+                    case HttpStatusCode.OK:
+                        this._logger.LogTrace("Response payload: {payload}", payload);
+                        QueryResult<DocumentType> documentTypesResult = JsonSerializer.Deserialize<QueryResult<DocumentType>>(payload);
+                        if (documentTypesResult != null)
+                        {
+                            retVal.Status = ExternalResultStatus.Success;
+                            retVal.Payload = documentTypesResult;
+                        }
+                        else
+                        {
+                            retVal.Status = ExternalResultStatus.Error;
+                            retVal.Message = "The response is empty";
+                        }
+
+                        break;
+                    case HttpStatusCode.NoContent:
+                        retVal.Status = ExternalResultStatus.Success;
+                        retVal.Message = "No content found";
+                        break;
+                    case HttpStatusCode.Forbidden:
+                        retVal.Status = ExternalResultStatus.Error;
+                        retVal.Message = "Forbidden";
+                        break;
+                    default:
+                        retVal.Status = ExternalResultStatus.Error;
+                        retVal.Message = $"Unable to contact endpoint {endpoint}. Http status {response.StatusCode}";
+                        break;
+                }
+            }
+            catch (Exception e)
```

Comment:
Is catching Exception considered an antipattern the way it is in Java?

Reply:
It would be if you catch and then re-throw without adding anything. On this case is adding information to the reply wrapper